src/module-5/double-slider/index.js

```diff
@@ -1,15 +1,157 @@
+      this.selected.to = this.min + Math.round((event.clientX - rect.left) / rect.width * (this.max - this.min));
+    }
+    this.update();
+  };
+
+  progressStop = () => {
+    this.moving.toLeft = false;
+    this.moving.toRight = false;
+    this.dispatchRangeEvent();
+  }
+
   constructor({
-    min = 100,
-    max = 200,
-    formatValue = value => value,
-    selected = {
-      from: min,
-      to: max
-    },
-    precision = 0,
-    filterName = ''
-  } = {}) {
-    // ... your logic
+                min = 100,
+                max = 200,
+                value = '',
+                formatValue = value => value,
+                selected = {
+                  from: min,
+                  to: max
+                },
+                precision = 0,
+                filterName = 'Price'
+              } = {}) {
+    this.min = min;
+    this.max = max;
+    this.formatValue = formatValue(value);
+    this.selected = selected;
+    this.precision = precision;
+    this.filterName = filterName;
+
+    this.addFormatValue();
+    this.countLeftValue();
+    this.countRightValue();
+
+    this.render();
+    this.getSubElements();
+    this.addEventListeners();
+  }
+
+  countLeftValue() {
+    return (this.selected.from - this.min) / (this.max - this.min) * 100 + '%';
+  }
+
+  countRightValue() {
+    return (this.max - this.selected.to) / (this.max - this.min) * 100 + '%';
+  }
+
+  addFormatValue(valueSlider) {
+    return this.formatValue + valueSlider;
+  }
+
+  get template() {
+    const {from, to} = this.selected;
+    return `<div class="range-slider" data-element="rangeSlider">
+        <span data-element="from">${this.addFormatValue(from)}</span>
+        <div class="range-slider__inner" data-element="sliderInner">
+          <span data-element="progress" class="range-slider__progress"
+          style="left: ${this.countLeftValue()}; right: ${this.countRightValue()}"></span>
+          <span data-element="thumbLeft" class="range-slider__thumb-left"
+          style="left: ${this.countLeftValue()}"></span>
+          <span data-element="thumbRight" class="range-slider__thumb-right"
+          style="right: ${this.countRightValue()}"></span>
+        </div>
+        <span data-element="to">${this.addFormatValue(to)}</span>
+      </div >`
+  }
+
+  render() {
+    const wrapper = document.createElement('div');
+    wrapper.innerHTML = this.template;
+    this.element = wrapper.firstElementChild;
+  }
+
+  getSubElements() {
+    const result = {};
+    const elements = this.element.querySelectorAll('[data-element]');
+    for (const subElement of elements) {
+      const name = subElement.dataset.element;
+      result[name] = subElement;
+    }
+    this.subElements = result;
+  }
+
+  remove() {
+    if (this.element) {
+      this.removeEventListeners();
+      this.element.remove();
+    }
+  }
+
+  destroy() {
+    this.remove();
+    this.element = null;
+  }
+
+  update() {
+    if (this.selected.from < this.min) {
```

Comment:
you may simplify this code using `Math.min/Math.max` methods, e.g.  
`this.selected.from = Math.min(Math.max(this.selected.from, this.min), this.selected.to)`  
take maximum of min and current value, then pick minimum between this value and to-value;